aspnetcore-features/ValidatingFileUploadExtension/ValidatingFileUploadExtension/FileValidator.cs

```diff
@@ -0,0 +1,20 @@
+﻿using ValidatingFileUploadExtension;
+
+public static class FileValidator
+{
+   
+    private static readonly List<FileFormatDescriptor> AllowedFormats = [new Image(), new Pdf()];
+
+    public static Result Validate(IFormFile file)
+    {
+        var fileExtension = Path.GetExtension(file.FileName);
```

Comment:
Use AsSpan so we call the ReadOnlySpan version. This was already mentioned in the first round of feedback